utils/algorithms.cpp

```diff
@@ -34,3 +36,16 @@ void sort(Vector<Shape*>& vector, Comparator cmp)
         std::swap(vector[min], vector[i]);
     }
 }
+
+// Moves the text stream up to the character specified
+void getTo(QTextStream &qin, const QChar &target)
+{
+    QChar buffer;
+
+    // Read in characters into the buffer until the buffer character is a whitespace
+    do
```

Comment:
Should be a ``while`` loop so that we can check for ``!qin.atEnd()`` before we iterate.

Reply:
It isn't necessary to write another line. ``QChar()`` constructs it with a ``\0`` so it isn't uninitialized. Therefore ``buffer != target`` is totally valid.

Reply:
aaaah I forgot about the constructor.  I was thinking of QChar as standard char that always starts uninitialized

Reply:
Oh wait but isn't this also a problem if the calling method wants the text buffer to get to the next instance of `\0` in the text stream?  Then with the pre-check while loop it would exit thinking it had gotten to the correct part of the text stream when in actuality the initial value of QChar and the target character just happened to be the same by coincidence